sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -994,6 +994,18 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
       }
     }
 
+    object ExtractStringAttribute {
+      @scala.annotation.tailrec
+      def unapply(expr: Expression): Option[Attribute] = {
+        expr match {
+          case attr: Attribute => Some(attr)
+          case Cast(child, dt: IntegralType, _, _)
+            if child.dataType.isInstanceOf[StringType] => unapply(child)
```

Comment:
This assumes Spark and Hive have the same behavior for implicit cast. Is it always true? For example, cast string to int may fail in Spark with ansi mode on, does hive have the same behavior?